packages/blade/src/components/BaseMenu/BaseMenuItem.tsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Box } from '~components/Box';
+import BaseBox from '~components/Box/BaseBox';
+import { Checkbox } from '~components/Checkbox';
+import { Text } from '~components/Typography';
+import { size } from '~tokens/global';
+import { getMediaQuery, makeBorderSize, makeSize } from '~utils';
+import { getFocusRingStyles } from '~utils/getFocusRingStyles';
+import { makeAccessible } from '~utils/makeAccessible';
+import type { BladeElementRef } from '~utils/types';
+
+const StyledMenuItemContainer = styled.button<{ color?: 'negative' }>((props) => {
+  return {
+    borderWidth: makeBorderSize(props.theme.border.width.none),
+    display: 'flex',
+    flexDirection: 'column',
+    textAlign: 'left',
+    backgroundColor: 'transparent',
+    padding: makeSize(props.theme.spacing[2]),
+    borderRadius: makeSize(props.theme.border.radius.medium),
+    marginTop: makeSize(props.theme.spacing[1]),
+    marginBottom: makeSize(props.theme.spacing[1]),
+    textDecoration: 'none',
+    cursor: 'pointer',
+    width: '100%',
+    [`@media ${getMediaQuery({ min: props.theme.breakpoints.m })}`]: {
+      padding: makeSize(props.theme.spacing[3]),
+    },
+    '&:focus-visible': getFocusRingStyles({ theme: props.theme }),
+    '&:hover:not([aria-disabled=true]), &.has-submenu-open': {
+      backgroundColor:
+        props.color === 'negative'
+          ? props.theme.colors.interactive.background.negative.faded
+          : props.theme.colors.interactive.background.gray.default,
+    },
+    '&[aria-selected=true]': {
+      backgroundColor: props.theme.colors.interactive.background.primary.faded,
+    },
+    '&[aria-selected=true]:hover': {
+      backgroundColor: props.theme.colors.interactive.background.primary.fadedHighlighted,
+    },
+  };
+});
+
+type BaseMenuItemProps = {
+  as?: 'a' | 'button';
+  title: string;
+  description?: string;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+  leading?: React.ReactNode;
+  trailing?: React.ReactNode;
+  titleSuffix?: React.ReactNode;
+  selectionType?: 'single' | 'multiple';
+  color?: 'negative';
+  className?: string;
+  href?: string;
+};
+
+const menuItemTitleColor = {
+  negative: {
+    default: 'feedback.text.negative.intense',
+    disabled: 'interactive.text.gray.disabled',
+  },
+  normal: { default: 'interactive.text.gray.normal', disabled: 'interactive.text.gray.disabled' },
+} as const;
+
+const menuItemDescriptionColor = {
+  default: 'interactive.text.gray.muted',
+  disabled: 'interactive.text.gray.disabled',
+} as const;
+
+const _BaseMenuItem: React.ForwardRefRenderFunction<BladeElementRef, BaseMenuItemProps> = (
+  {
+    title,
```

Comment:
On figma I can see isLoading state as well. Lets add that?

Reply:
Didn't see loading state anywhere in figma 🤔